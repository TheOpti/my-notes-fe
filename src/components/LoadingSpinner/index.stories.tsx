import * as React from 'react';
import { storiesOf } from '@storybook/react'

import LoadingSpinner from './'

storiesOf('LoadingSpinner', module)
  .add('All examples', () => {
    return (
      <div>
        <div style={{ margin: 10 }}> 
          <p>Large:</p>
          <LoadingSpinner size="large" />
        </div>
        <div style={{ margin: 10 }}> 
          <p>Medium:</p>
          <LoadingSpinner size="medium" />
        </div>
        <div style={{ margin: 10 }}> 
          <p>Small:</p>
          <LoadingSpinner size="small" />
        </div>
        <div style={{ margin: 10 }}> 
          <p>Large, monocolor:</p>
          <LoadingSpinner size="large" monocolor />
        </div>
        <div style={{ margin: 10 }}> 
          <p>Medium, monocolor:</p>
          <LoadingSpinner size="medium" monocolor />
        </div>
        <div style={{ margin: 10 }}> 
          <p>Small, monocolor:</p>
          <LoadingSpinner size="small" monocolor />
        </div>
      </div>
    );
  });