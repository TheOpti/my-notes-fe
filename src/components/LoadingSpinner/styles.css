.spinner {
  animation: rotator 1.4s linear infinite;
}

@keyframes rotator {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(270deg); }
}

.path {
  stroke-dasharray: 200;
  transform-origin: center;
}

.colors {
  animation: dash 1.4s ease-in-out infinite, colors 5.6s ease-in-out infinite
}

.monocolor {
  animation: dash 1.4s ease-in-out infinite;
  stroke: var(--main-dark-gray-color);
}

@keyframes colors {
	0% { stroke: var(--main-blue-color); }
	25% { stroke: var(--secondary-blue-color); }
	50% { stroke: var(--main-dark-gray-color); }
	75% { stroke: var(--main-red-color); }
  100% { stroke: var(--main-green-color); }
}

@keyframes dash {
  0% { 
    stroke-dashoffset: 200; 
  }
  50% {
    stroke-dashoffset: 50;
    transform: rotate(135deg);
  }
  100% {
    stroke-dashoffset: 200;
    transform: rotate(450deg);
  }
}