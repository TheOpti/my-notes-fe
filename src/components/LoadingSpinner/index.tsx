import React from 'react';
import cx from 'classnames';
import styles from './styles.css';

const sizes = {
  small: '24px',
  medium: '48px',
  large: '64px',
}

type LoadingSpinnerPropsType = {
  size: 'small' | 'medium' | 'large',
  monocolor?: boolean,
}
const LoadingSpinner: React.SFC<LoadingSpinnerPropsType> = (props: LoadingSpinnerPropsType) => {
  const { size, monocolor } = props;

  const spinnerSize = sizes[size];

  const circleClasses = cx({
    [styles.colors]: !monocolor,
    [styles.monocolor]: monocolor,
  }, [styles.path])

  return (
    <svg 
      className={styles.spinner} 
      width={spinnerSize}
      height={spinnerSize}
      viewBox="0 0 66 66" 
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle 
        className={circleClasses} 
        fill="none" 
        strokeWidth="6" 
        strokeLinecap="round"
         cx="33" cy="33" r="30"
      />
    </svg>
  );
}

LoadingSpinner.defaultProps = {
  size: 'medium',
  monocolor: false,
}

export default LoadingSpinner;