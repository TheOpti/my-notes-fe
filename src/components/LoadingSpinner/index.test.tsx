import React from "react";
import { mount } from 'enzyme';
import LoadingSpinner from './';

describe('LoadingSpinner', () => {
  it('should render correctly without any issues', () => {
    const wrapper = mount(<LoadingSpinner size="medium" />);

    expect(wrapper).toMatchSnapshot();
  });
});