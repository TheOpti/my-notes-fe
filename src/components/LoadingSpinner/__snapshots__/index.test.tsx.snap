// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner should render correctly without any issues 1`] = `
<LoadingSpinner
  monocolor={false}
  size="medium"
>
  <svg
    className="spinner"
    height="48px"
    viewBox="0 0 66 66"
    width="48px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      className="colors path"
      cx="33"
      cy="33"
      fill="none"
      r="30"
      strokeLinecap="round"
      strokeWidth="6"
    />
  </svg>
</LoadingSpinner>
`;
